src/uwtools/drivers/driver.py

```diff
@@ -77,8 +90,66 @@ def schema_file(self) -> str:
         The path to the file containing the schema to validate the config file against.
         """
 
+    @staticmethod
+    def stage_files(
+        run_directory: str, files_to_stage: Dict[str, str], link_files: bool = False
+    ) -> None:
+        """
+        Takes in run directory and dictionary of file names and paths that need to be staged in the
+        run directory.
+
+        Creates dst file in run directory and copies or links contents from the src path provided.
+        """
+
+        link_or_copy = os.symlink if link_files else shutil.copyfile
+
+        for dst_fn, src_path in files_to_stage.items():
+            dst_path = os.path.join(run_directory, dst_fn)
+            if isinstance(src_path, list):
+                self.stage_files(
+                    run_directory,
+                    {os.path.join(dst_path, os.path.basename(src)): src
+                        for src in src_path},
+                    link_files,
+                    )
+                continue
+            link_or_copy(src_path, dst_path)
+            msg = f"File {src_path} staged in run directory at {dst_fn}"
+            logging.info(msg)
+
+
     # Private methods
 
+    @staticmethod
+    def _create_dictable_configure_file(config_class: Config, config_values: dict, output_path:
```

Comment:
I don't necessarily think that a private method is mutually exclusive with being static. The way I understand it, the static part means that it doesn't reference any of the data in the object, so doesn't need a `self` argument. The private part means that we don't want to expose it, which I landed on given that we're imposing a Driver-specific YAML structure to provide the input data. 

In this case, the callers are the `FV3Forecast` methods `create_namelist()`, `create_model_configure()`, and `create_field_table()` who all have Schema-checked subsections that are held to the standard of having `base_file` and `update_values` entries.  It seems dangerous to let any caller use this method with any old unvalidated `config_values` dict.

re: config class. I have no idea! I have not yet checked in on what the type checker wants there. I was going out on a limb that if it had the right methods that matched it to the abstract base class, then it would work. I will circle back to this conversation when I find the answer. :)

Reply:
> I don't necessarily think that a private method is mutually exclusive with being static.

I think I agree. I think I confused more Java-esque public/private semantics, which would actually make it impossible to call a private method from outside the class itself -- which would make it being static useless -- with Python's "please don't" semantics. A good test might be whether `pylint` complains about accessing a private member at the call site; if we have to `# pylint: disable` every such call, I think we might be going against the grain and should consider dropping the privacy `_`, though your worry about a public method being called inappropriately stands (but is probably kind of true about every public method).